dace/codegen/targets/fpga.py

```diff
@@ -1221,3 +1236,35 @@ def generate_host_function_boilerplate(self, sdfg, state, kernel_name,
         for arr_node in nested_global_transients:
             self._dispatcher.dispatch_allocate(sdfg, state, None, arr_node,
                                                None, host_code_stream)
+
+    def _generate_Tasklet(self, *args, **kwargs):
+        # Call CPU implementation with this code generator as callback
+        self._cpu_codegen._generate_Tasklet(*args, codegen=self, **kwargs)
+
+    def define_out_memlet(self, sdfg, state_dfg, state_id, src_node, dst_node,
+                          edge, function_stream, callsite_stream):
+        self._dispatcher.dispatch_copy(src_node, dst_node, edge, sdfg,
+                                       state_dfg, state_id, function_stream,
+                                       callsite_stream)
+
+    def process_out_memlets(self, *args, **kwargs):
+        # Call CPU implementation with this code generator as callback
+        self._cpu_codegen.process_out_memlets(*args, codegen=self, **kwargs)
+
+    def generate_tasklet_preamble(self, *args, **kwargs):
+        # Fall back on CPU implementation
+        self._cpu_codegen.generate_tasklet_preamble(*args, **kwargs)
+
+    def generate_tasklet_postamble(self, sdfg, dfg, state_id, node,
+                                   function_stream, before_memlets_stream,
+                                   after_memlets_stream):
+        # Inject dependency pragmas on memlets
+        for edge in dfg.out_edges(node):
+            datadesc = sdfg.arrays[edge.data.data]
+            if (isinstance(datadesc, dace.data.Array)
+                    and (datadesc.storage == dace.StorageType.FPGA_Local
+                         or datadesc.storage == dace.StorageType.FPGA_Registers)
+                    and not cpp.is_write_conflicted(dfg, edge)):
+                self.generate_no_dependence_post(edge.src_conn,
```

Comment:
Nice and clean! 👍 
Shouldn't this be in xilinx.py though? (not sure OpenCL has a no_dependence pragma equivalent?)

Reply:
I chose to make this generic, so we have `no_dependence_pre` and `no_dependence_post` (symmetric for all the pragmas), so I wanna keep it this way for consistency :-) OpenCL has dependence pragmas, but before the code rather than after.